auth-web/src/views/auth/SigninView.vue

```diff
@@ -52,58 +52,44 @@ export default class Signin extends Mixins(NextPageMixin) {
 
   @Prop({ default: 'bcsc' }) idpHint: string
   @Prop({ default: '' }) redirectUrl: string
+  @Prop({ default: '' }) redirectUrlLoginFail: string
 
   private async mounted () {
     if (this.redirectUrl) {
-      this.setRedirectAfterLoginUrl(`/${this.redirectUrl}`)
+      this.setRedirectAfterLoginUrl(decodeURIComponent(this.redirectUrl))
     } else {
       this.setRedirectAfterLoginUrl(this.idpHint === 'idir' ? '/searchbusiness' : '')
     }
 
     // Initialize keycloak session
     const kcInit = await this.userStore.initKeycloak(this.idpHint)
     await new Promise((resolve, reject) => {
-      kcInit.success(async authenticated => {
-        if (authenticated) {
-          this.initializeSession()
-          // Make a POST to the users endpoint if it's bcsc (only need for BCSC)
-          if (this.idpHint === 'bcsc') {
-            await this.syncUserProfile()
-            // eslint-disable-next-line no-console
-            console.info('[SignIn.vue]Logged in User.Starting refreshTimer')
-            var self = this
-            let tokenService = new TokenService()
-            tokenService.initUsingUrl(`${process.env.VUE_APP_PATH}config/kc/keycloak.json`).then(function (success) {
-              tokenService.scheduleRefreshTimer()
-            })
+      kcInit
+        .success(async authenticated => {
+          if (authenticated) {
+            this.initializeSession()
+            // Make a POST to the users endpoint if it's bcsc (only need for BCSC)
+            if (this.idpHint === 'bcsc') {
+              await this.syncUserProfile()
+              // eslint-disable-next-line no-console
+              console.info('[SignIn.vue]Logged in User.Starting refreshTimer')
+              var self = this
+              let tokenService = new TokenService()
+              tokenService.initUsingUrl(`${process.env.VUE_APP_PATH}config/kc/keycloak.json`).then(function (success) {
+                tokenService.scheduleRefreshTimer()
+              })
+            }
+            resolve()
           }
-          resolve()
-        }
-      })
+        })
+        .error(() => {
+          if (this.redirectUrlLoginFail) {
```

Comment:
If the user clicks on cancel in BC Services Card login, the it will take the user back to home page as it's a pre-configured thing in BCSC. Not sure, when this block will be executed.

Reply:
Oh okay, I didn't realize that.  I suppose this wouldn't be fired at all then, unless KeyCloak was unavailable or something.  Should I remove?

Reply:
I think we can keep the code there, won't do any harm.